src/Package/Impl/DataInspect/VariableViewModel.cs

```diff
@@ -4,41 +4,53 @@
 using System;
 using System.Collections.Generic;
 using System.Diagnostics;
-using System.Linq;
 using System.Threading.Tasks;
 using System.Windows.Input;
 using Microsoft.Common.Core;
 using Microsoft.R.Debugger;
 using Microsoft.R.Editor.Data;
 using Microsoft.VisualStudio.PlatformUI;
 using Microsoft.VisualStudio.R.Package.DataInspect.Office;
-using Microsoft.VisualStudio.R.Package.Utilities;
+using Microsoft.VisualStudio.R.Package.Shell;
 
 namespace Microsoft.VisualStudio.R.Package.DataInspect {
     /// <summary>
     /// Model for variable tree grid, that provides UI customization of <see cref="DebugEvaluationResult"/>
     /// </summary>
-    public sealed class EvaluationWrapper : RSessionDataObject, IIndexedItem {
+    public sealed class VariableViewModel : RSessionDataObject, IIndexedItem {
+        private IObjectDetailsViewer _detailsViewer;
+        private string _title;
 
-        public EvaluationWrapper() { Index = -1; }
+        public VariableViewModel() { Index = -1; }
 
         /// <summary>
-        /// Create new instance of <see cref="EvaluationWrapper"/>
+        /// Create new instance of <see cref="VariableViewModel"/>
         /// </summary>
         /// <param name="evaluation">R session's evaluation result</param>
         /// <param name="truncateChildren">true to truncate children returned by GetChildrenAsync</param>
-        public EvaluationWrapper(DebugEvaluationResult evaluation, int index = -1, int? maxChildrenCount = null) :
+        public VariableViewModel(DebugEvaluationResult evaluation, int index = -1, int? maxChildrenCount = null) :
             base(evaluation, maxChildrenCount) {
 
             Index = index;
+            var result = DebugEvaluation as DebugValueEvaluationResult;
+            if (result != null) {
+                SetViewButtonStatus(result);
+            }
+        }
 
-            CanShowDetail = ComputeDetailAvailability(DebugEvaluation as DebugValueEvaluationResult);
+        private void SetViewButtonStatus(DebugValueEvaluationResult result) {
+            var aggregator = VsAppShell.Current.ExportProvider.GetExportedValue<IObjectDetailsViewerAggregator>();
+            _detailsViewer = aggregator.GetViewer(result);
+            _title = result.Name;
+
+            CanShowDetail = _detailsViewer != null;
             if (CanShowDetail) {
-                ShowDetailCommand = new DelegateCommand(ShowVariableGridWindowPane, (o) => CanShowDetail);
+                ShowDetailCommand = new DelegateCommand(async (o) => await _detailsViewer.ViewAsync(result.Expression, _title), (o) => CanShowDetail);
```

Comment:
`DelegateCommand` doesn't have proper ctor for async methods. You should either call `ViewAsync` with `DoNotWait`, or just provide a plain method and use it directly instead of command. Also, for `DelegateCommand` you need to call `RaiseCanExecuteChanged` manually :(